### ISSUE TITLE
Nvidia driver installation fails

### ISSUE THREAD

**Issue by levingerdes** [esa-prl/rover_docker#6]
Since commit 9946c0346dc357ed9572f7749dc3cfb1a565d00f , which addressed issue #3 , the setup step `sh run_container.sh` produces an error, because we're now trying to install a package without root/sudo permissions:

```
Step 27/27 : RUN DEBIAN_FRONTEND=noninteractive apt-get install -y nvidia-driver-410
 ---> Running in 925b212c846c
E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
The command '/bin/sh -c DEBIAN_FRONTEND=noninteractive apt-get install -y nvidia-driver-410' returned a non-zero code: 100
```

Executing the command before switching to a non-privileged user or by invoking it via `sudo`, we are informed that the package is not available under that name:
```
Step 27/28 : RUN DEBIAN_FRONTEND=noninteractive sudo apt-get install -y nvidia-driver-410
 ---> Running in e41bc696b138
Reading package lists...
Building dependency tree...
Reading state information...
Package nvidia-driver-410 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'nvidia-driver-410' has no installation candidate
The command '/bin/sh -c DEBIAN_FRONTEND=noninteractive sudo apt-get install -y nvidia-driver-410' returned a non-zero code: 100
```

**Comment by maxehrhardt** [comment]
Adding an `sudo apt-get update` in the same row as the install could solve the problem. Since Docker always uses the cache, that could be too old.

**Comment by levingerdes** [comment]
Update: The specific driver was not found, only newer versions: `nvidia-driver-440` can be found but entails a lot of different packages and reconfigurations. One of them is `keyboard-configuration` which asks for user input to specify the keyboard layout and then gets stuck. I assume that the combination of `noninteractive` and a fallback to `Readline` input frontend as opposed to `Dialog` might cause a problem here.

**Comment by wieset** [comment]
I think the reason that keyboard-configuration is stopping for user input is because the `DEBIAN_FRONTEND` variable is not  available in the sudo environment. Moving sudo to the front should fix this:

`cmd="RUN sudo DEBIAN_FRONTEND=noninteractive apt-get update && sudo apt-get install -y $driver"`

**Comment by wieset** [comment]
@levingerdes Which Ubuntu version are you running as your host system? My guess is you have `nvidia-driver-410` installed there but it's not available by default on Ubuntu 18.04+. Ideas to solve this would be to add the graphics-drivers ppa or to install a newer driver version inside the container and see if that works as well.

**Comment by levingerdes** [comment]
> @levingerdes Which Ubuntu version are you running as your host system? My guess is you have `nvidia-driver-410` installed there but it's not available by default on Ubuntu 18.04+. Ideas to solve this would be to add the graphics-drivers ppa or to install a newer driver version inside the container and see if that works as well.

I'm running 18.04 and I've upgraded my driver, which led to the keyboard configuration issue.

> I think the reason that keyboard-configuration is stopping for user input is because the DEBIAN_FRONTEND variable is not available in the sudo environment. Moving sudo to the front should fix this: `cmd="RUN sudo DEBIAN_FRONTEND=noninteractive apt-get update && sudo apt-get install -y $driver"`

Ah, that was it, makes sense. Thanks!
